feature/search/src/main/java/com/google/samples/apps/nowinandroid/feature/search/SearchScreen.kt

```diff
@@ -128,26 +305,29 @@ private fun SearchTextField(onSearchQueryChanged: (String) -> Unit) {
             )
         },
         trailingIcon = {
-            IconButton(onClick = { textState.value = "" }) {
+            IconButton(onClick = {
+                searchQuery.value = ""
+                onSearchQueryChanged("")
+            }) {
                 Icon(
                     imageVector = NiaIcons.Close,
                     contentDescription = stringResource(
-                        id = searchR.string.clear_search_text,
+                        id = searchR.string.clear_search_text_content_desc,
                     ),
                     tint = MaterialTheme.colorScheme.onSurface,
                 )
             }
         },
         onValueChange = {
-            textState.value = it
+            searchQuery.value = it
```

Comment:
Use viewModel.searchQuery as single source of truth

```suggestion
```

Reply:
Done